### ISSUE TITLE
Modal focus box appearing incorrectly in New Arch app

### ISSUE THREAD

**Issue by Yajur-Grover** [microsoft/react-native-windows#14485]
### Problem Description

In New Arch Gallery, when opening a Modal example, focus is on the Modal example but the visual focus border is in the wrong place on the screen, which causes confusion on where the user is currently navigated to. 

Screen Recording: https://microsoft-my.sharepoint-df.com/:v:/p/yajurgrover/ETSlhbTE705PtzkftCRlP4oBzft9tcBobyaklgo5T2CnOA?e=7c46aA

Screenshot:
![Image](https://github.com/user-attachments/assets/7ad7e2f3-c323-4379-805b-fd27d11277d4)

### Steps To Reproduce

1. Launch New Arch Gallery
2. Go to the Modal example page
3. Open some of the example modals
4. Tab onto the opened modal

### Expected Results

The black focus border should be around where the focus of the keyboard actually is.

### CLI version

npx @react-native-community/cli -v

### Environment

```markdown
npx @react-native-community/cli info
```

### Target Platform Version

None

### Visual Studio Version

None

### Build Configuration

None
